app/src/main/java/org/cru/godtools/ui/dashboard/tools/ToolsFragment.kt

```diff
@@ -2,24 +2,118 @@ package org.cru.godtools.ui.dashboard.tools
 
 import android.os.Bundle
 import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.ConcatAdapter
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import org.ccci.gto.android.common.androidx.lifecycle.onDestroy
 import org.ccci.gto.android.common.sync.swiperefreshlayout.widget.SwipeRefreshSyncHelper
 import org.cru.godtools.R
 import org.cru.godtools.databinding.DashboardToolsFragmentBinding
+import org.cru.godtools.download.manager.GodToolsDownloadManager
 import org.cru.godtools.fragment.BasePlatformFragment
+import org.cru.godtools.model.Tool
+import org.cru.godtools.model.Translation
+import org.cru.godtools.ui.tooldetails.startToolDetailsActivity
+import org.cru.godtools.ui.tools.ToolsAdapter
+import org.cru.godtools.ui.tools.ToolsAdapterCallbacks
+import org.cru.godtools.ui.tools.ToolsAdapterViewModel
 
 @AndroidEntryPoint
-class ToolsFragment : BasePlatformFragment<DashboardToolsFragmentBinding>(R.layout.dashboard_tools_fragment) {
+class ToolsFragment :
+    BasePlatformFragment<DashboardToolsFragmentBinding>(R.layout.dashboard_tools_fragment),
+    ToolCategoriesAdapter.Callbacks,
+    ToolsAdapterCallbacks {
+    @Inject
+    internal lateinit var downloadManager: GodToolsDownloadManager
 
-    private val dataModel: ToolsFragmentDataModel by viewModels()
-    override fun onBindingCreated(binding: DashboardToolsFragmentBinding, savedInstanceState: Bundle?) {
-        super.onBindingCreated(binding, savedInstanceState)
-        binding.refresh.setupSwipeRefresh()
-        binding.hasSpotlight = dataModel.hasSpotlight
+    // region Data Model
+    private val toolsCategoryDataModel: ToolsCategoryDataModel by viewModels()
```

Comment:
```suggestion
    private val dataModel: ToolsFragmentDataModel by viewModels()
```

Reply:
the datamodel for this fragment should be renamed to actually match the fragment name